File: backend/src/features/room/model/response/roomMember.go
package response

// ResRoomMember 방 멤버 정보
type ResRoomMember struct {
	UserID          uint   `json:"user_id"`
	UserName        string `json:"user_name"`
	Email           string `json:"email"`
	ProfileImageURL string `json:"profile_image_url"`
	Permission      string `json:"permission"` // READ_ONLY, READ_WRITE, OWNER
	JoinedAt        int64  `json:"joined_at"`
}

// ResRoomMembers 방 멤버 목록 응답
type ResRoomMembers struct {
	RoomID  uint            `json:"room_id"`
	Members []ResRoomMember `json:"members"`
}

File: frontend/app/src/main/java/com/dailymemo/domain/models/RoomMember.kt
package com.dailymemo.domain.models

data class RoomMember(
    val userId: Long,
    val userName: String,
    val email: String,
    val profileImageUrl: String?,
    val permission: RoomPermission,
    val joinedAt: Long
)

enum class RoomPermission {
    OWNER,
    READ_WRITE,
    READ_ONLY;

    companion object {
        fun fromString(value: String): RoomPermission {
            return when (value.uppercase()) {
                "OWNER" -> OWNER
                "READ_WRITE" -> READ_WRITE
                "READ_ONLY" -> READ_ONLY
                else -> READ_ONLY
            }
        }
    }
}
